File: golang/uPIMulator/src/linker/kernel/instruction/cc/sub_nz_cc.go
package cc

import (
	"errors"
)

type SubNzCc struct {
	condition Condition
}

func (this *SubNzCc) Init(condition Condition) {
	conditions := map[Condition]bool{
		C:    true,
		NC:   true,
		Z:    true,
		NZ:   true,
		XZ:   true,
		XNZ:  true,
		OV:   true,
		NOV:  true,
		MI:   true,
		PL:   true,
		EQ:   true,
		NEQ:  true,
		SPL:  true,
		SMI:  true,
		GES:  true,
		GEU:  true,
		GTS:  true,
		GTU:  true,
		LES:  true,
		LEU:  true,
		LTS:  true,
		LTU:  true,
		XGTS: true,
		XGTU: true,
		XLES: true,
		XLEU: true,
		TRUE: true,
	}

	if _, found := conditions[condition]; !found {
		err := errors.New("condition is not allowed")
		panic(err)
	}

	this.condition = condition
}

func (this *SubNzCc) Condition() Condition {
	return this.condition
}

File: python_cpp/uPIMulator_frontend/src/abi/isa/cc/sub_nz_cc.py
from typing import Set

from abi.isa.cc._base_cc import BaseCC
from abi.isa.instruction.condition import Condition


class SubNZCC(BaseCC):
    def __init__(self, condition: Condition):
        super().__init__(
            SubNZCC.conditions(), condition,
        )

    @staticmethod
    def conditions() -> Set[Condition]:
        return {
            Condition.C,
            Condition.NC,
            Condition.Z,
            Condition.NZ,
            Condition.XZ,
            Condition.XNZ,
            Condition.OV,
            Condition.NOV,
            Condition.MI,
            Condition.PL,
            Condition.EQ,
            Condition.NEQ,
            Condition.SPL,
            Condition.SMI,
            Condition.GES,
            Condition.GEU,
            Condition.GTS,
            Condition.GTU,
            Condition.LES,
            Condition.LEU,
            Condition.LTS,
            Condition.LTU,
            Condition.XGTS,
            Condition.XGTU,
            Condition.XLES,
            Condition.XLEU,
            Condition.TRUE,
        }
